src/libraries/System.Text.RegularExpressions/tests/RegexCultureTests.cs

```diff
@@ -29,7 +29,7 @@ public void CharactersComparedOneByOne_AnchoredPattern(string pattern, string in
             }
         }
 
-        [Theory]
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotInvariantGlobalization))]
```

Comment:
hm this test says it's comparing things with invariant culture so I'd have expected it to work with invariant mode. how does it fail?

Reply:
Perhaps only 2 of the 4 InlineData's need the conditionalizing?

Reply:
Actually, all four were failing with 
```
Assert.StartsWith() Failure:
Expected: Ö‍
Actual:   Ö
```

Reply:
So it's not invariant globalization, it's some other bug? 

as `Assert.StartsWith(S1, S2, StringComparison.InvariantCulture);` should not be affected by invariant setting.

Reply:
Right, it wasn't immediately clear what the fix is to me, so I created an issue for now.

Reply:
I think `InvariantMode` and `StringComparison.InvariantCulture` have nothing to do with each other. When we're running on `InvariantMode` the string comparison is different, it is done char by char and basically it does `if (string1[i] - string2[i] != 0) return string1[i] - string2[i]`

In this case, those chars are indeed different, so substracting them will give a result different than 0, hence the test failing. We should just skip this test on Invariant globalization.

https://source.dot.net/#System.Private.CoreLib/CompareInfo.cs,448

Which will cause the comparison to instead call `string.SequenceCompareTo` which will endup doing the comparison by chars here:
https://source.dot.net/#System.Private.CoreLib/SpanHelpers.Char.cs,114

cc: @tarekgh

Reply:
@safern is right. Invariant string comparison is still a linguistic comparison which is not supported by *Globalization invariant mode*. so any code using Invariant culture comparison and using codepoints outside ASCII range is expected can fail when running in Invariant mode.

Reply:
TIL.. thanks

Reply:
If you're curious why the output shows the same strings see this discussion: https://github.com/dotnet/runtime/issues/39480#issuecomment-659766921